pkg/agent/cniserver/server_windows.go

```diff
@@ -47,9 +47,20 @@ func (s *CNIServer) hostNetNsPath(netNS string) string {
 }
 
 // isInfraContainer return if a container is infra container according to the network namespace path.
-// On Windows platform, the network namespace of infra container is "none".
+// On Windows platform:
+//   - If use Docker as CRI runtime, the network namespace of infra container is "none".
+//   - If use ContainerD as CRI runtime, the network namespace of infra container is
+//     a string which does not contains ":".
 func isInfraContainer(netNS string) bool {
-	return netNS == infraContainerNetNS
+	return netNS == infraContainerNetNS || !strings.Contains(netNS, ":")
+}
+
+// isDockerContainer return if a container is created by Docker with the provided network namespace.
+// The network namespace format of Docker container is:
+//   - Infra container: "none"
+//   - Workload container: "container:$infra_container_id"
+func isDockerContainer(netNS string) bool {
+	return netNS == "none" || strings.Contains(netNS, ":")
```

Comment:
so just to confirm:
* for docker: first we have netNS set to `none` (infra container), then netNS set to a string including `:` for workload containers
* for containerd: single netNS which is a string *not* including `:`

is that correct?

Reply:
Yes, correct!